### ISSUE TITLE
HealthKit import crashes after a few seconds

### ISSUE THREAD

**Issue by patricks** [bryceco/GoMap#723]
I have a really big HealthKit database (2000+ workouts with locations). If I tap the Import button the app crashes after a few seconds. After looking at the source code I assume the problem ist that all routes are imported directly. I would suggest do use a selection first where you could select a workout. IMHO only one workout is required at once?

**Comment by bryceco** [comment]
Yes it does import everything. I don't use HealthKit myself so I don't have an easy way to test large datasets. 

An alternative to having a selection screen is to just import the last 30 days. What do you think?

**Comment by patricks** [comment]
That would be a easy solution without building extra UI.

**Comment by bryceco** [comment]
Awesome! Thanks for the feedback.
